### PR TITLE
Fixes an issue where placing a bid hangs

### PR DESCRIPTION
**Pull request by yuki24**
fixes https://artsyproduct.atlassian.net/browse/AUCT-499
blocked by https://github.com/artsy/emission/pull/1741

eThis is the actual fix for the bug. Basically the issue is that `result.position.id` was not replaced even after we switched to MP v2. As a result of this, `undefined` is passed on to the polling query which causes the server to return an error. This PR replaces all the occurances of `id` in the `<ConfirmBid>` component.

I have another PR that adds more type checks to mitigate the same type of risks: https://github.com/artsy/emission/compare/master...add-more-type-checks

### CODE REVIEW

File: src/lib/Components/Bidding/Screens/ConfirmBid.tsx [hunk 1]
```diff
@@ -64,14 +64,14 @@ const MAX_POLL_ATTEMPTS = 20
 const queryForBidPosition = (bidderPositionID: string) => {
```
Comment (dleve123) [inline src/lib/Components/Bidding/Screens/ConfirmBid.tsx]: while we're here, thoughts on renaming to queryforBid**der**Position as a `BidderPosition` is a thing, but a `BidPosition` isn't a thing?

File: src/lib/Components/Bidding/Screens/ConfirmBid.tsx [hunk 2]
```diff
@@ -64,14 +64,14 @@ const MAX_POLL_ATTEMPTS = 20
...
   return metaphysics({
     query: `
-      {
+      query ConfirmBidBidderPositionQuery($bidderPositionID: String!) {
```
Comment (dleve123) [inline src/lib/Components/Bidding/Screens/ConfirmBid.tsx]: out of curiosity, can you provide some more context for this change? Why name the query?

File: src/lib/Components/Bidding/Screens/__tests__/ConfirmBid-tests.tsx [hunk 3]
```diff
@@ -803,7 +811,7 @@ const mockRequestResponses = {
           message_header: "Success",
           message_description_md: "",
           position: {
-            internalID: "some-bidder-position-id",
+            internalID: "bidder-position-id-from-mutation",
```
Comment (dleve123) [inline src/lib/Components/Bidding/Screens/__tests__/ConfirmBid-tests.tsx]: what's the level of effort requested to refactor the `pollingForBid` fixture to use relay like this fixture? Could we resolve these TODOs in this PR? Seems like that would increase the robustness of these tests for any future refactorings

File: src/lib/Components/Bidding/Screens/ConfirmBid.tsx (hunk 2 above)
Comment (yuki24) [reply src/lib/Components/Bidding/Screens/ConfirmBid.tsx]: This used to be a dynamic query due to the presence of the string interpolation, which halts static analysis done by GraphQL. I've changed it to use `variables` to inject `bidderPositionID`. The query name could be any name, but I followed Relayjs' convention of `ComponentNameQueryName(Query|Mutation)`.

File: src/lib/Components/Bidding/Screens/__tests__/ConfirmBid-tests.tsx (hunk 3 above)
Comment (yuki24) [reply src/lib/Components/Bidding/Screens/__tests__/ConfirmBid-tests.tsx]: The expectations for polling was simply missing and this basically backfills test coverage. The first exception specifically tests that the `internalID` from the mutation results is properly passed on to the first polling request:

https://github.com/artsy/emission/blob/ec0e94d6403122496c4ac3eb226378d16588377b/src/lib/Components/Bidding/Screens/ConfirmBid.tsx#L297

And the second one specifically tests the handling of the one from the polling results:

https://github.com/artsy/emission/blob/ec0e94d6403122496c4ac3eb226378d16588377b/src/lib/Components/Bidding/Screens/ConfirmBid.tsx#L317

While in production both calls would return the exact same id, I've used two different ids in test to make sure each one of the calls works as expected.

Comment (dleve123) [reply src/lib/Components/Bidding/Screens/__tests__/ConfirmBid-tests.tsx]: This lead to a IRL conversation about the feasibility about generating types for these fixtures. See this comment: https://github.com/artsy/emission/pull/1744/files#r306406212

TL;DR is that we aren't going to worry about this type enhancement in this PR.
